app/helpers/medication_refill_helper.rb

```diff
@@ -3,21 +3,16 @@
 module MedicationRefillHelper
   include CalendarHelper
   # Save refill date to Google calendar
-  # rubocop:disable RescueStandardError
   def save_refill_to_google_calendar(medication)
     return true unless current_user.google_oauth2_enabled? &&
                        new_cal_refill_reminder_needed?(medication)
-
-    begin
-      args = calendar_uploader_params(medication)
-      CalendarUploader.new(args).upload_event
-    rescue
-      return_to_sign_in
-    else
-      true
-    end
+    args = calendar_uploader_params(medication)
+    CalendarUploader.new(args).upload_event
+  rescue Google::Apis::ClientError
+    return_to_sign_in
+  rescue Google::Apis::ServerError
+    redirect_to_medication(medication)
```

Comment:
I removed the recursive call here because we don't know how long the server will be down. And because this is not a background job, there is no need to halt everything else and wait for it to pass